tests/providers/apache/impala/hooks/test_impala.py

```diff
@@ -0,0 +1,113 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import unittest
+from unittest import mock
+from unittest.mock import patch
+
+from airflow.models import Connection
+from airflow.providers.apache.impala.hooks.impala import ImpalaHook
+
+
+class TestImpalaHookConn(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.connection = Connection(
+            login="login",
+            password="password",
+            host="host",
+            port=21050,
+            schema="test",
+        )
+
+        class UnitTestImpalHook(ImpalaHook):
+            conn_name_attr = "impala_conn_id"
+
+        self.db_hook = UnitTestImpalHook()
+        self.db_hook.get_connection = mock.Mock()
+        self.db_hook.get_connection.return_value = self.connection
+
+    @patch("airflow.providers.apache.impala.hooks.impala.connect")
+    def test_get_conn(self, mock_connect):
+        self.db_hook.get_conn()
+        mock_connect.assert_called_once_with(
+            host="host", port=21050, user="login", password="password", database="test"
+        )
+
+
+class TestImpalaHook(unittest.TestCase):
```

Comment:
Would be cool to write it as pytest
We have on going effort to do it so we can have less work in the future https://github.com/apache/airflow/pull/26831

Reply:
yeah sure

Reply:
done

Reply:
Class still based on `unittest.TestCase` and use their methods.
You need to:
1. Remove bases of `unittest.TestCase`
2. Rename `def setUp(self)` to `def setup_method`. And you don't need call `super()`
3. Instead of `self.assertEqual(a, b)` use just use `assert a == b`